### DISCUSSION TITLE
Is there a reconnected callback?

### DISCUSSION THREAD

**Discussion by theogravity** [enisdenjo/graphql-ws#312] [Q&A] [answered]
We have a case where we want to do something when the client reconnects from a connection being lost, I don't see one in the `on` definitions for it in `createClient()`

**Comment by enisdenjo** [comment, marked as answer]
There's no need for a specific event. Here's how you can perform the check:
```ts
let disconnected = false;
createClient({
  ...
  on: {
    closed: () => {
      disconnected = true;
    },
    connected: () => {
      if (disconnected) {
        // reconnected
      }
    },
  },
});
```
